lib/ipam/ipam_block_reader_writer.go

```diff
@@ -284,6 +292,11 @@ func (rw blockReaderWriter) releaseBlockAffinity(ctx context.Context, host strin
 		return errBlockClaimConflict{Block: b}
 	}
 
+	// Don't release block affinity if we require it to be empty and it's not empty.
+	if requireEmpty && !b.empty() {
+		return nil
```

Comment:
Let's add a debug log here so we know when we're hitting this branch.

Reply:
addressed in a808da3